pydm/widgets/spinbox.py

```diff
@@ -61,13 +79,20 @@ def widget_ctx_menu(self):
         QMenu or None
             If the return of this method is None a new QMenu will be created by `assemble_tools_menu`.
         """
-        def toggle():
+        def toggle_step():
             self.showStepExponent = not self.showStepExponent
 
+        def toggle_write():
+            self.writeOnPress = not self.writeOnPress
+
         menu = self.lineEdit().createStandardContextMenu()
         menu.addSeparator()
         ac = menu.addAction('Toggle Show Step Size')
-        ac.triggered.connect(toggle)
+        ac.triggered.connect(toggle_step)
+
+        ac_write = menu.addAction('Toggle write on press')
```

Comment:
Great addition to the context menu! I like the ability to be able to change that at runtime as well. 👍